shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -624,8 +626,18 @@ private void initializePlatformViewIfNeeded(int viewId) {
 
     PlatformView platformView = factory.create(context, viewId, createParams);
     View view = platformView.getView();
-    platformViews.put(viewId, view);
 
+    // Reverse scale the view based on the screen density.
+    //
+    // Flow pixels are based on the physical resolution. For example, 1000 pixels in flow equals
+    // 500 points in Android if the display density is 2.0.
+    float density = getDisplayDensity();
+    view.setPivotX(0);
+    view.setPivotY(0);
+    view.setScaleX(1 / density);
+    view.setScaleY(1 / density);
```

Comment:
@cyanglaz fyi. This is the equivalent of https://github.com/flutter/engine/blob/81489ece5396a78be0996d4310062e70a8d0cdfa/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L396-L405